flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/casting/MapAndMultisetToStringCastRule.java

```diff
@@ -101,8 +109,16 @@ protected String generateCodeBlockInternal(
             String returnVariable,
             LogicalType inputLogicalType,
             LogicalType targetLogicalType) {
-        final LogicalType keyType = ((MapType) inputLogicalType).getKeyType();
-        final LogicalType valueType = ((MapType) inputLogicalType).getValueType();
+        boolean isMap = inputLogicalType instanceof MapType;
```

Comment:
No need to have a separate variable for this, just put it in the condition. Also please use the `is(LogicalTypeFamily)` method for checking if it's a map or not